### PR TITLE
Handles err connection refused

### PR DESCRIPTION
**Pull request by aljones15**

### CODE REVIEW

File: tests/10-client-api.spec.common.cjs [hunk 1]
```diff
@@ -31,6 +31,43 @@ describe('http-client API', () => {
     err.requestUrl.should.equal('http://httpbin.org/status/404');
     err.response.status.should.equal(404);
   });
+  it('handles a connection refused error', async () => {
+    let err;
+    let response;
+    // the intention here is to use an used http port
+    // the port used can not be higher than 65535 making it illegal
+    const nonExistentResource = 'http://localhost:65535';
```
Comment (dlongley) [inline tests/10-client-api.spec.common.cjs:39]: Can we make this `https` since it's the more common scheme?

**Review by dlongley — CHANGES_REQUESTED** [review]
Thanks! A few nits. Looks like we're getting the connection refused error here which is good. We'll need to figure out why we don't get it elsewhere. Could be related to sharing an HTTP(S) agent or another problem at a higher level in the stack.

File: tests/10-client-api.spec.common.cjs [hunk 2]
```diff
@@ -31,6 +31,43 @@ describe('http-client API', () => {
...
+    should.exist(
+      err.requestUrl,
+      'Expected nonExistentResource "err.requestUrl" to exist.'
+    );
+    err.requestUrl.should.equal(
+      nonExistentResource,
+      `Expected nonExistentResource "err.requestUrl" to be ` +
+        `${nonExistentResource}`
+    );
+    // node 18's global fetch seems to be changing the error return type
```
Comment (davidlehn) [inline tests/10-client-api.spec.common.cjs:63]: "seems to"?  please use a more factual comment that says what is happening in node 18 vs others.

**Comment by aljones15** [comment]
just so this is known: this appears to be a duplicate test: https://github.com/digitalbazaar/http-client/blob/586ee0e8044a4139047e27ec89236155a46f0087/tests/10-client-api.spec.common.cjs#L204-L221

Another test already did the same thing.

File: tests/10-client-api.spec.common.cjs (hunk 2 above)
Comment (aljones15) [reply tests/10-client-api.spec.common.cjs]: I changed the language here: https://github.com/digitalbazaar/http-client/pull/29/commits/f7bd52d5eadc92cc5ea9c512b6b9caf1b2665c41 but the weakness of the assertion accurately represents the amount of research done. Node 18 introduced global fetch, it turns out: https://github.com/nodejs/undici/blob/c7f527b2a727a726d9e53c19297314d2850717cb/lib/fetch/response.js#L104-L107

it takes the original error and assigns it to `cause`. I'm still not 100% sure that is why in node 18 we start getting `code` in `cause`, but it seems likely as the reason.

**Review by dlongley — APPROVED** [review]
Thanks! LGTM, @davidlehn?

File: karma.conf.cjs [hunk 3]
```diff
@@ -6,7 +6,7 @@ module.exports = function(config) {
   config.set({
     // base path that will be used to resolve all patterns (eg. files, exclude)
     basePath: '',
-
+    captureConsole: true,
```
[the remark this answers is not in the document]
Comment (aljones15) [reply karma.conf.cjs]: it's actually true by default, but I went ahead and removed it here: https://github.com/digitalbazaar/http-client/pull/29/commits/d2fe1e6ce6e897d4872cdfa6ccebbe0607b5f7ea

**Pull request merged**